crates/types/src/config/worker.rs

```diff
@@ -124,10 +137,86 @@ impl Default for WorkerOptions {
             max_command_batch_size: NonZeroUsize::new(32).expect("Non zero number"),
             snapshots: SnapshotsOptions::default(),
             trim_delay_interval: None,
+            durability_mode: None,
+            experimental_partition_driven_log_trimming: false,
         }
     }
 }
 
+#[serde_as]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, derive_more::Display, Serialize, Deserialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "schemars", schemars(rename = "DurabilityMode"))]
```

Comment:
Does the rename attribute have an effect given that the enum is also called `DurabilityMode`?